pkg/controllers/nodeclaim/lifecycle/controller.go

```diff
@@ -85,7 +85,7 @@ func NewController(clk clock.Clock, kubeClient client.Client, cloudProvider clou
 
 func (c *Controller) Register(_ context.Context, m manager.Manager) error {
 	return controllerruntime.NewControllerManagedBy(m).
-		Named(c.Name()).
+		Named(c.name()).
```

Comment:
I'm not sure that I quite understand why this needs to be done -- if a struct has more methods that aren't defined on an interface, that doesn't mean that it doesn't fulfill the interface. This seems like a change that Intellij should take up vs. one that we should make in _our_ code

Reply:
I just found out that this type of error occurs inconsistently in rare cases, and yes, I understood that this doesn't really go against the go interface rules. I'll let Jetbrains know about this. Thank you!